File: ShowDown/Player.js
class Player{
    constructor(name){
        this.name = name;
    }

    static async NameHimself(playerNumber) {
        const players = [];
        for (let i = 0; i < playerNumber; i++) {
            const playerName = CommandLine(`Enter name for Player ${i+1}:`);
            const player = new HumanPlayer(playerName);
            players.push(player);
            console.log(`P${i+1}'s name is ${playerName}`);
        }
        return players;
    }
}

class HumanPlayer extends Player{
    constructor(name) {
        super(name);
    }
}

class AIPlayer extends Player{
    constructor(name) {
        super(name);
    }
}

module.exports = { Player, HumanPlayer, AIPlayer };

File: ShowDown_Java/Player.java
package ShowDown_Java;

public abstract class Player {
    protected String name; 
    
    // constructor
    public Player() {
        
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public abstract void nameHimself();

    // without this, will show 
    // Players: [ShowDown_Java.HumanPlayer@28d93b30, ShowDown_Java.AIPlayer@1b6d3586, 
    //           ShowDown_Java.AIPlayer@4554617c, ShowDown_Java.AIPlayer@74a14482]
    @Override
    public String toString() {
        return name; 
    }
}
